drivers/ad7746/ad7746.c

```diff
@@ -0,0 +1,288 @@
+        goto release;
+    }
+
+    if (dev->params.vt_mode != AD7746_VT_MD_DIS) {
+        /* Enable voltage / temperature channel and set mode */
+        reg = (1 << AD7746_VT_SETUP_VTEN_BIT) |
+              (dev->params.vt_mode << AD7746_VT_SETUP_VTMD0_BIT);
+        if (i2c_write_reg(I2C, ADDR, AD7746_REG_VT_SETUP, reg, 0)) {
+            DEBUG("[ad7746] init - error: unable to enable the v/t channel\n");
+            goto release;
+        }
+    }
+
+    /* Set EXC sources */
+    reg = (dev->params.exc_config << AD7746_EXC_SETUP_INV_EXCA_BIT);
+    if (i2c_write_reg(I2C, ADDR, AD7746_REG_EXC_SETUP, reg, 0)) {
+        DEBUG("[ad7746] init - error: unable to set EXC outputs\n");
+        goto release;
+    }
+
+    /* Set DAC A capacitance value */
+    if (dev->params.dac_a_cap) {
+        assert(dev->params.dac_a_cap <= AD7746_DAC_MAX);
+        reg = (1 << AD7746_DACAEN_BIT) | dev->params.dac_a_cap;
+        if (i2c_write_reg(I2C, ADDR, AD7746_REG_CAP_DAC_A, reg, 0)) {
+            DEBUG("[ad7746] init - error: unable to set DAC A\n");
+            goto release;
+        }
+    }
+
+    /* Set DAC B capacitance value */
+    if (dev->params.dac_b_cap) {
+        assert(dev->params.dac_b_cap <= AD7746_DAC_MAX);
+        reg = (1 << AD7746_DACBEN_BIT) | dev->params.dac_b_cap;
+        if (i2c_write_reg(I2C, ADDR, AD7746_REG_CAP_DAC_B, reg, 0)) {
+            DEBUG("[ad7746] init - error: unable to set DAC B\n");
+            goto release;
+        }
+    }
+
+    /* Set to continuous mode and configured sample rates */
+    reg = (1 << AD7746_CONFIGURATION_MD0_BIT) |
+          (dev->params.cap_sample_rate << AD7746_CONFIGURATION_CAPF0_BIT) |
+          (dev->params.vt_sample_rate << AD7746_CONFIGURATION_VTF0_BIT);
+    if (i2c_write_reg(I2C, ADDR, AD7746_REG_CONFIGURATION, reg, 0)) {
+        DEBUG("[ad7746] init - error: unable to set mode and SR\n");
+        goto release;
+    }
+
+    res = AD7746_OK;
+
+release:
+    i2c_release(I2C);
+    return res;
+}
+
+int ad7746_set_vt_channel_mode(const ad7746_t *dev, ad7746_vt_mode_t mode)
+{
+    uint8_t reg;
+    int res = AD7746_NOI2C;
+
+    assert(dev);
+    i2c_acquire(I2C);
+    if (i2c_read_reg(I2C, ADDR, AD7746_REG_VT_SETUP, &reg, 0)) {
+        goto release;
+    }
+
+    if (mode == AD7746_VT_MD_DIS) {
+        reg &= ~(1 << AD7746_VT_SETUP_VTEN_BIT);
+    }
+    else {
+        /* Enable voltage / temperature channel and set mode */
+        reg &= ~((1 << AD7746_VT_SETUP_VTMD0_BIT) |
+                 (1 << AD7746_VT_SETUP_VTMD1_BIT));
+        reg |= (1 << AD7746_VT_SETUP_VTEN_BIT) |
+               (mode << AD7746_VT_SETUP_VTMD0_BIT);
+    }
+
+    if (i2c_write_reg(I2C, ADDR, AD7746_REG_VT_SETUP, reg, 0)) {
+        DEBUG("[ad7746] set_vt_ch - error: unable to set v/t channel mode\n");
+        goto release;
+    }
+
+release:
+    i2c_release(I2C);
+    return res;
+}
+
+int ad7746_read_raw_ch(const ad7746_t *dev, uint8_t ch, uint32_t *raw)
```

Comment:
To avoid code duplication (as reading raw data is really similar for both channels) I did this function and implemented an specific one for each channel as inline in ad7746.h (`ad7746_read_raw_cap_ch` and `ad7746_read_raw_vt_ch`)